dask_sql/physical/rex/core/call.py

```diff
@@ -115,6 +115,25 @@ def __init__(self, operation: Callable, unary_operation: Callable = None):
 
     def reduce(self, *operands, **kwargs):
         if len(operands) > 1:
+            # Doing math against dates requires a Timedelta
+            # Find all the operands that are of a datetime64 type
+            date_operands = [
+                idx for idx in {0, 1} if pd.api.types.is_datetime64_dtype(operands[idx])
+            ]
+
+            # If there are datetime64 operands we need to make sure that the other operands
+            # in the list are Timedelta for the operation to work.
+            if date_operands:
+                # Operands is a Set, since we are altering it must convert to a List
+                operands = list(operands)
+
+                # Knowing there are datetime types in the operands check for incompatable other types
+                # If found, convert them to Timedelta
+                for idx, operand in enumerate(operands):
+                    # Default to `Day`/`D` since that is what PostgreSQL does
+                    if isinstance(operand, (np.int64, str)):
+                        operands[idx] = np.timedelta64(operand, "D")
```

Comment:
Following up on our internal sync:

```suggestion
                    if isinstance(operand, np.int64):
                        operands[idx] = np.timedelta64(operand, "D")
                    elif isinstance(operand, str):
                        operands[idx] = np.datetime64(operand)
```